### PR TITLE
Add a list of supported storage for artifact store

### PR DESCRIPTION
**Pull request by nabenabe0928**
## Motivation

As it is not clear what storage is supported by artifact store objects, I added the list.

## Description of the changes

- Add the list of supported storage in doc string

### CODE REVIEW

**Comment by y0z** [comment]
@c-bata, @porink0424 Could you review this PR?

File: docs/source/reference/artifacts.rst [hunk 1]
```diff
@@ -5,6 +5,17 @@ optuna.artifacts
 
 The :mod:`~optuna.artifacts` module provides the way to manage artifacts (output files) in Optuna.
 Please check :ref:`artifact_tutorial` for the instruction.
+The storages covered by :mod:`~optuna.artifacts` are the following:
+
++-------------------------+-----------------------------------------------------------------+
+|                         |                        Supported Storage                        |
```
Comment (porink0424) [inline docs/source/reference/artifacts.rst:11]: How about adding this:
```suggestion
|      Class Name      |                        Supported Storage                        |
```

Comment (nabenabe0928) [reply docs/source/reference/artifacts.rst]: Thank you for the review, updated

**Review by porink0424 — APPROVED** [review]
LGTM!

**Pull request merged**